pkg/build/api/validation/validation.go

```diff
@@ -647,3 +649,20 @@ func validatePostCommit(spec buildapi.BuildPostCommitSpec, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+func validateRuntimeImage(sourceStrategy *buildapi.SourceBuildStrategy, fldPath *field.Path) (allErrs field.ErrorList) {
+	if sourceStrategy.RuntimeImage == nil {
+		return
+	}
+
+	if sourceStrategy.RuntimeImage.Name == "" {
+		allErrs = append(allErrs, field.Invalid(fldPath, sourceStrategy.RuntimeImage, "the runtime image name cannot be empty"))
+		return
+	}
+
+	if sourceStrategy.RuntimeImage.Name != "" && len(sourceStrategy.RuntimeArtifacts) == 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, sourceStrategy.RuntimeArtifacts, "at least one runtime artifacts mapping must be specified"))
+		return
```

Comment:
Just realized that we probably shouldn't do this check: when `RuntimeArtifacts` is empty, S2I will try to extract this mapping from the `io.openshift.s2i.assemble-input-files` label on a runtime image.